File: Typescript/Practice/easy/easy_olderThanMe.ts
#!/usr/bin/env npx ts-node

class Person {
  name: string;
  age: number;

  constructor (name: string, age: number) {
    this.name = name;
    this.age = age;
  };

  compareAge (per: Person) {
    if (this.age < per.age) {
      console.log(`${per.name} is older than me.`);
    } else if (this.age > per.age) {
      console.log(`${per.name} is younger than me.`);
    } else {
      console.log(`${per.name} is the same age as me.`);
    };
  };
};

const p1 = new Person("Samuel", 24);
const p2 = new Person("Joel", 36);
const p3 = new Person("Lily", 24);

p1.compareAge(p2);
p2.compareAge(p1);
p1.compareAge(p3);

File: Typescript/Practice/26072022/easy/easy_olderThanMe.py
#!/usr/bin/env python

class Person:
  def __init__(self, name, age):
    self.name = name
    self.age = age

  def compare_age(self, per):
    if per.age > self.age:
      print(f'{per.name} is older than me.')
    elif per.age < self.age:
      print(f'{per.name} is younger than me.')
    else:
      print(f'{per.name} is the same age as me.')

p1 = Person("Samuel", 24)
p2 = Person("Joel", 36)
p3 = Person("Lily", 24)

p1.compare_age(p2)
p2.compare_age(p1)
p1.compare_age(p3)
